forc-pkg/src/manifest.rs

```diff
@@ -263,13 +265,20 @@ impl Manifest {
             || self.pkg_dep(STD).is_some()
             || self.dep(STD).is_some()
         {
+            if !include_implicit_std {
+                println_yellow_err("Warning: You specified an STD path but set `implicit-std` to `false`.\
+                 Your custom std path will take precdence and be imported regardless.");
+            }
             return;
         }
         // Add a `[dependencies]` table if there isn't one.
         let deps = self.dependencies.get_or_insert_with(Default::default);
         // Add the missing dependency.
-        let std_dep = implicit_std_dep(sway_git_tag.to_string());
-        deps.insert(STD.to_string(), std_dep);
+        if !include_implicit_std {
+            println_yellow_err("Warning: Project being built without an STD");
+            let std_dep = implicit_std_dep(sway_git_tag.to_string());
+            deps.insert(STD.to_string(), std_dep);
+        }
```

Comment:
Rather than adding a `if` branch here, would you mind if we added the condition to the list of early return conditions above (L266)? *See the other comment with the code suggestion.*

Also just a heads up - we *do* want to include `std` implicitly if `include_implicit_std` is `true`.